### PR TITLE
Further improvements applied to tracking codes (regex filters applied…

### PR DESCRIPTION
**Pull request by Krystian-gnm**
## What does this change?
Further improvements applied to tracking codes (regex filters applied + '.emailjson' and '.emailtxt' removed from path URL when links are passed through to Braze)
## Screenshots
N/A
## What is the value of this and can you measure success?
To provide security improvement on potential hack injections + removing of the '.emailjson' and 'emailtxt' from path URL when email is passed through to Braze
## Checklist
N/A
### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)
N/A
### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/master/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)
N/A
### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Comment by paulbrown1982** [comment]
LGTM 👍 but please get someone from DotCom team to review too. Thanks.

File: common/app/views/fragments/email/support.scala.html [hunk 1]
```diff
@@ -1,5 +1,7 @@
 @(page: model.Page)(implicit request: RequestHeader)
 
+@componentIdPart = @{ if (page.metadata.isFront) request.path.toLowerCase.replace("/", "_").replaceFirst(".email(json|txt)", "").filter(_.toString.matches("[a-z_]")) else "_email_article" }
```
Comment (AWare) [inline common/app/views/fragments/email/support.scala.html]: Can this be pulled out into a helper function? I'm not sure what `componentIdPart` means.

**Review by AWare — APPROVED** [review]
Looks good, can you rebase as @shtukas changes are also showing up here.

**Review by SiAdcock — COMMENTED** [review]
Look okay to me, but can we rebase to get rid of those changes to package.json and yarn.lock?

Edit: just spotted @aware made the same comment, apologies for the duplicate!

**Comment by Krystian-gnm** [comment]
Closed in favour of https://github.com/guardian/frontend/pull/21198

**Pull request closed without merging**